crates/diesel_models/src/payment_intent.rs

```diff
@@ -516,34 +501,43 @@ pub struct PaymentIntentUpdateFields {
 
 // TODO: uncomment fields as necessary
 #[cfg(feature = "v2")]
-#[derive(Clone, Debug, AsChangeset, router_derive::DebugAsDisplay)]
+#[derive(Clone, Debug, AsChangeset, router_derive::DebugAsDisplay, Serialize, Deserialize)]
 #[diesel(table_name = payment_intent)]
 pub struct PaymentIntentUpdateInternal {
-    // pub amount: Option<MinorUnit>,
-    // pub currency: Option<storage_enums::Currency>,
     pub status: Option<storage_enums::IntentStatus>,
-    // pub amount_captured: Option<MinorUnit>,
-    // pub customer_id: Option<common_utils::id_type::CustomerId>,
-    // pub return_url: Option<>,
-    // pub setup_future_usage: Option<storage_enums::FutureUsage>,
-    // pub metadata: Option<pii::SecretSerdeValue>,
-    pub modified_at: PrimitiveDateTime,
     pub active_attempt_id: Option<common_utils::id_type::GlobalAttemptId>,
-    // pub description: Option<String>,
-    // pub statement_descriptor: Option<String>,
-    // #[diesel(deserialize_as = super::OptionalDieselArray<pii::SecretSerdeValue>)]
-    // pub order_details: Option<Vec<pii::SecretSerdeValue>>,
-    // pub attempt_count: Option<i16>,
+    pub modified_at: PrimitiveDateTime,
+    pub amount: Option<MinorUnit>,
+    pub currency: Option<storage_enums::Currency>,
+    pub shipping_cost: Option<MinorUnit>,
+    pub skip_external_tax_calculation: Option<bool>,
+    pub surcharge_applicable: Option<bool>,
+    pub surcharge_amount: Option<MinorUnit>,
+    pub tax_on_surcharge: Option<MinorUnit>,
+    pub routing_algorithm_id: Option<common_utils::id_type::RoutingId>,
+    pub capture_method: Option<common_enums::CaptureMethod>,
+    pub authentication_type: Option<common_enums::AuthenticationType>,
+    pub billing_address: Option<Encryption>,
+    pub shipping_address: Option<Encryption>,
+    pub customer_present: Option<bool>,
+    pub description: Option<common_utils::types::Description>,
+    pub return_url: Option<common_utils::types::Url>,
+    pub setup_future_usage: Option<storage_enums::FutureUsage>,
+    pub apply_mit_exemption: Option<bool>,
+    pub statement_descriptor: Option<common_utils::types::StatementDescriptor>,
+    pub order_details: Option<Vec<masking::Secret<OrderDetailsWithAmount>>>,
+    pub allowed_payment_method_types: Option<pii::SecretSerdeValue>,
+    pub metadata: Option<pii::SecretSerdeValue>,
+    pub connector_metadata: Option<pii::SecretSerdeValue>,
+    pub feature_metadata: Option<FeatureMetadata>,
+    pub enable_payment_link: Option<bool>,
+    pub payment_link_config: Option<PaymentLinkConfigRequestForPayments>,
+    pub request_incremental_authorization: Option<RequestIncrementalAuthorization>,
+    pub session_expiry: Option<PrimitiveDateTime>,
+    pub frm_metadata: Option<pii::SecretSerdeValue>,
+    // Setting it to common_enums::External3dsAuthenticationRequest gives error on `AsChangeset` derivation
```

Comment:
Does this error still exist?

Reply:
Its stored as a boolean in the DB, so a bool type is appropriate here. Removed the comment